### PR TITLE
Enhancement to allow usage with temporary security credentials (eg: assume role)

### PR DESCRIPTION
**Pull request by c0state**
Tries to support #217 in a backward compatible manner. Tests needed.

### CODE REVIEW

File: resources/configuration_file_template.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 s3_id: YOUR_AWS_S3_ACCESS_KEY_ID
 s3_secret: YOUR_AWS_S3_SECRET_ACCESS_KEY
+s3_token: YOUR_AWS_S3_SESSION_TOKEN # if using temporary credentials with a session token (eg: when assuming a role)
```
Comment (laurilehmijoki) [inline resources/configuration_file_template.yml]: This should probably be commented out by default, if we assume that most users will not use it. What do you think about this assumption?

Comment (c0state) [inline resources/configuration_file_template.yml]: Yes, this makes sense. I've updated the PR.

**Comment by laurilehmijoki** [comment]
Hello @c0state – in https://github.com/laurilehmijoki/s3_website/tree/c0state-allow_sts_tokens I've renamed `s3_token` to `session_token`. I think it's better to use the concept "session token", because that's what's in <http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/BasicSessionCredentials.html#getSessionToken-->, for example.

I also did some changes to the documantation.

What do you think of the changes?

**Comment by c0state** [comment]
Hi @laurilehmijoki - looks good to me. 

(I had used the `s3_` prefix on the session token variable to keep with the convention for the id and secret since the token is part of the credentials and also since the current `s3_id` and `s3_secret` aren't technically s3 specific either)

**Comment by laurilehmijoki** [comment]
I'll try to release this within a few days.

**Comment by laurilehmijoki** [comment]
Released in the version **3.1.0**. Thank you @c0state!

**Pull request merged**